src/main/java/uk/gov/hmcts/reform/bulkscan/orchestrator/services/ccd/EnvelopeReferenceCollectionHelper.java

```diff
@@ -0,0 +1,74 @@
+package uk.gov.hmcts.reform.bulkscan.orchestrator.services.ccd;
+
+import com.fasterxml.jackson.databind.ObjectMapper;
+import org.springframework.stereotype.Service;
+import org.springframework.util.CollectionUtils;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.model.ccd.CaseAction;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.model.ccd.CcdCollectionElement;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.model.ccd.EnvelopeReference;
+import uk.gov.hmcts.reform.bulkscan.orchestrator.services.config.ServiceConfigProvider;
+
+import java.util.List;
+import java.util.Map;
+
+import static java.util.Arrays.asList;
+import static java.util.stream.Collectors.toList;
+
+@Service
+public class EnvelopeReferenceCollectionHelper {
+
+    private final ObjectMapper objectMapper;
+    private final ServiceConfigProvider serviceConfigProvider;
+
+    public EnvelopeReferenceCollectionHelper(
+        ObjectMapper objectMapper,
+        ServiceConfigProvider serviceConfigProvider
+    ) {
+        this.objectMapper = objectMapper;
+        this.serviceConfigProvider = serviceConfigProvider;
+    }
+
+    public boolean serviceSupportsEnvelopeReferences(String service) {
+        return serviceConfigProvider.getConfig(service).getCaseDefinitionHasEnvelopeIds();
+    }
+
+    /**
+     * Appends envelope reference to the provided envelope references collection.
+     *
+     * <p>This collection represents bulkScanEnvelopes field in service case</p>
+     *
+     * @param existingEnvelopeReferences Current value of the field that holds envelope references in case,
+     *                                   in raw format (after deserialising case data as Object)
+     * @param envelopeId                 Id of the envelope to be appended
+     * @param action                     Action that the envelope caused on the case - create/update
+     * @return Updated collection, serialisable to CCD-compatible format
+     */
+    public List<CcdCollectionElement<EnvelopeReference>> appendEnvelopeReference(
```

Comment:
existingEnvelopeReferences and updatedReferences can they not be of same type ?
Also can we define a wrapper class say `EnvelopeReference` and which can consist of both fields existingReference and updatedReference ?

Reply:
just a thought let me know what you think not that too strong about it.

Reply:
Maybe I should reduce this method just to converting the collection and the appending would happen in the calling code?

Reply:
Done